src/modals/examples/modal-overlay.js

```diff
@@ -0,0 +1,138 @@
+ * @param {array} actionButtons array of button objects. Each button can have the following properties:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.label        - the text to display on the button
+ * <li>.class        - (optional) classes to add to the button
+ * <li>.actionFn     - (optional) use defined function to call when the button is clicked
+ * <li>.isDisabled   - (optional) boolean true if the button should be disabled by default
+ * <li>.isCancel     - (optional) boolean true is the button should cancel and dismiss the modal
+ * </ul>
+ *
+ * @example
+ <example module="patternfly.modals">
+
+ <file name="index.html">
+ <div ng-controller="DemoModalOverlayCtrl">
+    <button ng-click="open()" class="btn btn-default">Launch Modal Overlay</button>
+    <pf-modal-overlay is-open="isOpen"
+          on-close="onClose()"
+          modal-id="modalId"
+          modal-body-path="modalBodyPath"
+          modal-title="modalTitle"
+          action-buttons="actionButtons"></pf-modal-overlay>
+
+   <button ng-click="open2()" class="btn btn-default">Launch Second Modal Overlay</button>
+   <pf-modal-overlay is-open="isOpen2"
+         on-close="onClose2()"
+         modal-id="modalId2"
+         modal-body-path="modalBodyPath2"
+         modal-title="modalTitle2"
+         hide-close-icon="hideCloseIcon2"
+         backdrop-close="backdropClose"
+         title-id="titleId2"
+         action-buttons="actionButtons2"></pf-modal-overlay>
+ </div>
+ </file>
+
+ <file name="script.js">
+ angular.module('patternfly.modals').controller('DemoModalOverlayCtrl', function( $scope, $log ) {
+
+      // first example
+      $scope.open = function () {
+          $scope.isOpen = true;
+       };
+      $scope.onClose = function() {
+          $scope.isOpen = false;
+       };
+
+      $scope.modalId = "demoModal1";
+      $scope.modalTitle = "First Demo Title";
+      $scope.modalBodyPath = 'pf-modal-body.html';
+      $scope.actionButtons = [
+          {
+            label: "Cancel",
+            isCancel: true
+          },
+          {
+            label: "Save",
+            class: "btn-primary custom-class",
+            actionFn: function() {
+              $log.info("Save clicked");
+            }
+          }];
+
+      // second example
+      $scope.open2 = function () {
+          $scope.isOpen2 = true;
+       };
+      $scope.onClose2 = function() {
+          $scope.isOpen2 = false;
+       };
+
+      $scope.modalId2 = "demoModal2";
+      $scope.modalTitle2 = "Second Demo Title";
+      $scope.titleId2 = "demoTitle2";
+      $scope.modalBodyPath2 = 'pf-modal-body2.html';
+      $scope.hideCloseIcon2 = true;
+      $scope.backdropClose = true;
+      $scope.actionButtons2 = [
+          {
+            label: "Cancel",
+            isCancel: true
+          },
+          {
+            label: "OK",
+            class: "btn-primary"
+          }];
+ });
+
+ </file>
+
+ <file name="pf-modal-body.html">
```

Comment:
I guess I'm expecting to see this HTML in between `<pf-modal-overlay><contents of pf-modal-body.html></pf-modal-overlay>`, and have the `ng-transclude` attribute in the HTML template for the component.